### PR TITLE
Feature/HM-130-Admins-Add-heighborhoods-rating

### PR DESCRIPTION
**Pull request by petarbelchev**
Neighborhoods Rating functionality has been implemented.

### CODE REVIEW

File: src/Admins/Admins.Application/Features/Admins/Commands/AddNeighbourhoodsRating/AddNeighbourhoodsRatingCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+using MediatR;
+
+namespace BuildingMarket.Admins.Application.Features.Admins.Commands.AddNeighbourhoodsRating
+{
+    public class AddNeighbourhoodsRatingCommandHandler(IAdminRepository adminRepository) : IRequestHandler<AddNeighbourhoodsRatingCommand>
+    {
+        private readonly IAdminRepository _adminRepository = adminRepository;
+
+        public async Task Handle(AddNeighbourhoodsRatingCommand request, CancellationToken cancellationToken)
+            => await _adminRepository.AddNeighbourhoodsRating(request, cancellationToken);
```
Comment (rosteslav) [inline src/Admins/Admins.Application/Features/Admins/Commands/AddNeighbourhoodsRating/AddNeighbourhoodsRatingCommandHandler.cs]: repositories should not work based on commands, use `NeighbourhoodsRatingModel` we created in this review

File: src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs [hunk 2]
```diff
@@ -61,6 +65,47 @@ public async Task AddMultiplePropertiesFromCsvFile(IFormFile csvFile)
...
+                    Id = 1,
+                    ForLiving = JsonConvert.SerializeObject(command.ForLiving.First()),
+                    ForInvestment = JsonConvert.SerializeObject(command.ForInvestment.First()),
+                    Budget = JsonConvert.SerializeObject(command.Budget.First()),
+                    Luxury = JsonConvert.SerializeObject(command.Luxury.First())
+                };
+                var secondary = new NeighbourhoodsRating
+                {
+                    Id = 2,
+                    ForLiving = JsonConvert.SerializeObject(command.ForLiving.ElementAt(1)),
```
Comment (rosteslav) [inline src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs]: .Last(). ElementAt(1) is not a good way

File: src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs [hunk 3]
```diff
@@ -61,6 +65,47 @@ public async Task AddMultiplePropertiesFromCsvFile(IFormFile csvFile)
...
+                    ForInvestment = JsonConvert.SerializeObject(command.ForInvestment.ElementAt(1)),
+                    Budget = JsonConvert.SerializeObject(command.Budget.ElementAt(1)),
+                    Luxury = JsonConvert.SerializeObject(command.Luxury.ElementAt(1))
+                };
+
+                if (_context.NeighbourhoodsRating.Any(n => n.Id == best.Id))
```
Comment (rosteslav) [inline src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs]: better delete
`await _context.NeighbourhoodsRating.ExecuteDeleteAsync(cancellationToken);`
all and just insert

File: src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs [hunk 4]
```diff
@@ -78,6 +123,51 @@ public async Task<IEnumerable<IdentityUser>> GetAllBrokers()
             return Enumerable.Empty<IdentityUser>();
         }
 
+        public async Task<NeighbourhoodsRatingModel> GetNeighbourhoodsRating(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("DB get neighbourhoods rating");
+
+            try
+            {
+                var ratings = await _context.NeighbourhoodsRating.Where(n => n.Id == 1 || n.Id == 2).ToArrayAsync(cancellationToken);
```
Comment (rosteslav) [inline src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs]: just all, without where

File: src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs [hunk 5]
```diff
@@ -78,6 +123,51 @@ public async Task<IEnumerable<IdentityUser>> GetAllBrokers()
...
+                var result = new NeighbourhoodsRatingModel
+                {
+                    ForLiving = ratings.Select(r => JsonConvert.DeserializeObject<string[]>(r.ForLiving)),
```
Comment (rosteslav) [inline src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs]: isn't that `IEnumerable<IEnumerable<string>>`, why are you Deserializing it to <string[]>?
did you test it? Does it work?

**Review by rosteslav — CHANGES_REQUESTED** [review]

File: src/Admins/Admins.Application/Features/Admins/Commands/AddNeighbourhoodsRating/AddNeighbourhoodsRatingCommandHandler.cs (hunk 1 above)
Comment (petarbelchev) [reply src/Admins/Admins.Application/Features/Admins/Commands/AddNeighbourhoodsRating/AddNeighbourhoodsRatingCommandHandler.cs]: Got it

File: src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs (hunk 5 above)
Comment (petarbelchev) [reply src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs]: It is working, however there is no reason to be string[]

**Review by rosteslav — APPROVED** [review]

**Pull request merged**